### PR TITLE
CB-7827: Allow user to specify android activity name

### PR DESCRIPTION
**Pull request by cjpearson**
This change adds an attribute to config.xml android-activityName that can be used to specify a custom activity name. If none is specified it will still default to MainActivity. See http://markmail.org/message/pj5ig4b54nk4l7dr for a discussion of this issue.

### CODE REVIEW

File: cordova-lib/src/cordova/platform.js [hunk 1]
```diff
@@ -521,19 +521,27 @@ function getCreateArgs(platDetails, projectRoot, cfg, template_dir, opts) {
...
     var pkg = cfg.packageName().replace(/[^\w.]/g,'_');
     // CB-6992 it is necessary to normalize characters
     // because node and shell scripts handles unicode symbols differently
     // We need to normalize the name to NFD form since iOS uses NFD unicode form
     var name = platDetails.platform == 'ios' ? unorm.nfd(cfg.name()) : cfg.name();
     args.push(output, pkg, name);
+
+    var activityName = cfg.android_activityName();
+    if (activityName && platDetails.platform === 'android') {
+        activityName = activityName.replace(/W/g, '');
```
Comment (agrieve) [inline cordova-lib/src/cordova/platform.js]: why not allow any Ws? did you mean `\W`?

File: cordova-lib/src/cordova/platform.js [hunk 2]
```diff
@@ -521,19 +521,27 @@ function getCreateArgs(platDetails, projectRoot, cfg, template_dir, opts) {
...
+	args.push(activityName);
```
Comment (agrieve) [inline cordova-lib/src/cordova/platform.js]: We need this to continue to work for older versions of cordova-android. You should add this arg only if:

    semver.gte(platDetails.version, '4.0.0-dev')

File: cordova-lib/src/cordova/platform.js (hunk 1 above)
Comment (cjpearson) [inline cordova-lib/src/cordova/platform.js]: Yup, typo there.

File: cordova-lib/src/cordova/platform.js (hunk 2 above)
Comment (cjpearson) [inline cordova-lib/src/cordova/platform.js]: Thanks for the review. I'll add this.

**Comment by cjpearson** [comment]
@agrieve I've addressed your comments, could you take another look? Thanks!

**Comment by fredgalvao** [comment]
Shouldn't this be documented somewhere? It does not appear when I run `cordova help create` as a valid option.

**Comment by faenschi** [comment]
Should this work with cordova 4.3.1? Nothing happens if I add android-activityName="MyName" to config.xml

**Pull request closed without merging**